packages/common/src/stream.ts

```diff
@@ -118,6 +118,11 @@ export interface LogEntry {
   cid: CID
   type: CommitType
   timestamp?: number
+  // Holding on to these fields when fetching and processing new commits significantly reduces the number of IPFS
+  // lookups required. Use of these fields is currently limited to `conflict-resolution.ts` but further optimization of
+  // the code is possible by relaying them to code that looks the fields up again.
+  commit?: any
```

Comment:
I don't think we can put these into LogEntry as then we'll be keeping the contents of every commit around in the log within the state object for each stream, which could be a significant memory overhead. We need a separate structure that is just used for the log as we load and apply it, and then a smaller-subset of that data can go into the LogEntry's that go into the state object of the fully-loaded stream

Reply:
could go the other way and make LogEntry be what ConflictResolution uses during the load process and make a new type for what goes in the state object, but the main point is that these two things need to be different types so that we're not holding all the raw IPFS commit data in memory for every stream

Reply:
Yes, agreed about not storing the contents of all commits in the log. That won't actually happen at the moment because the two new fields (`commit` and `envelope`) are only populated and used during the application of a new `tip`, then discarded.

I was attempting to reduce IPFS lookups during tip application without changing adjacent code too much but if this seems like a hack then I can change the approach.

Reply:
Overloading `LogEntry` did seem a bit of a hack when I did it TBH but it was so tempting because the new structure would need all the same fields as `LogEntry` but one (`type: CommitType`) 😅

I wouldn't mind creating a new structure for loading/processing new commits. Something like the one you sent to me earlier:

```
interface CommitData {
  cid: CID
  commit: CeramicCommit
  envelope?: DagJWS
  timestamp?: number
}
```

Reply:
In other words, `fetchLog` (and other functions) would change to something like this:

```
export async function fetchLog(
  dispatcher: Dispatcher,
  cid: CID,
  stateLog: HistoryLog,
-  log: LogEntry[] = [],
+  newCommits: CommitData[] = [],
  timestamp?: number
): Promise<LogEntry[]> {
```

Reply:
I would then, however, have to translate `LogEntry` to `CommitData` because `snapshotAtCommit()` calls `applyLogToState()` using part of `StreamState.log` (array of `LogEntry`).

Reply:
interfaces can extend each other.  So maybe CommitData extends LogEntry and adds the `commit` and `envelope` fields.  Or maybe they both extend some smaller interface so LogEntry can add `type` without it being needed in `CommitData`

Reply:
How does this look @stbrody? I like `CommitData` extending `LogEntry` even if `type` isn't quite necessary in `CommitData`.